### PR TITLE
Add CPP codestyle checks to build

### PR DESCRIPTION
**Pull request by nickbattam-tessella**
Implement a standard set of static code style checks to the build.
Created as a new "analyse" build support additional rules in the future.

Report is uploaded and displayed in Jenkins. No rules for failing builds with an increasing number of warnings applied have been applied- these can bee added later if required.

### CODE REVIEW

**Comment by nickbattam-tessella** [comment]
clang-version is not available on the server; re implement with cppcheck; this is available on the build servers (https://anvil.softeng-support.ac.uk/help/guide/examples/cpptools/)

File: tools/build_config/build.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ function print_package_versions() {
   echo "$(cmake --version | head -n 1)"
   echo "Matlab: ${MATLAB_ROOT}"
   echo "$(g++ --version | head -n 1)"
+  echo "$(cppcheck --version | head -n 1)"
```
Comment (hsautessella) [inline tools/build_config/build.sh:30]: These commands don't need the `echo "${  }"`, just  `cppcheck --version | head -n 1` will achieve the same thing

File: tools/build_config/build.sh [hunk 2]
```diff
@@ -110,16 +128,20 @@ function main() {
     print_package_versions
   fi
 
+  if ((${analyze})); then
```
[the remark this answers is not in the document]
Comment (hsautessella) [reply tools/build_config/build.sh:131]: Just `if ((analyze)); then` works

File: tools/build_config/build.sh (hunk 1 above)
Comment (nickbattam-tessella) [reply tools/build_config/build.sh:30]: I left this for consistency with the rest of the file; the Herbert `build.sh` has had these fixes made on  a previous iteration -- I didn't want to make more changes than needed here as you're editing the same file in https://github.com/pace-neutrons/Horace/pull/210

File: tools/build_config/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (nickbattam-tessella) [reply tools/build_config/build.sh:131]: I left this for consistency with the rest of the file; the Herbert build.sh has had these fixes made on a previous iteration -- I didn't want to make more changes than needed here as you're editing the same file in #210

File: documentation/smg/08_setting_up_jenkins_ci.md [hunk 3]
```diff
@@ -81,7 +81,8 @@ and should be prefixed with `PR-` if the pipeline is building pull requests, e.g
         - `CMAKE_VERSION`: The version of CMake to load
         - `MATLAB_VERSION`: The (release) version of Matlab to load
         - `GCC_VERSION`: The version of GCC to use (Linux only)
-
+    - `CPPCHECK_VERSION`: The version of CppCheck to use (Linux only)
```
[the remark this answers is not in the document]
Comment (nickbattam-tessella) [reply documentation/smg/08_setting_up_jenkins_ci.md:84]: no

[the remark this answers is not in the document]
Comment (nickbattam-tessella) [reply documentation/smg/08_setting_up_jenkins_ci.md:84]: no

**Pull request merged**